In [ ]:

from google.colab import drive

In [4]:
import pandas as pd

In [ ]:
drive.mount("/content/drive")
%cd /content/drive/MyDrive/ma

In [7]:
sales_data = pd.read_csv('sales_data_updated.csv')

#### Виведи перші 5 рядків датасету.

In [8]:
sales_data.head()

order_date      time  aging  customer_id gender device_type  \
0  2018-01-01  10:11:40    5.0        29317   Male         Web   
1  2018-01-01  22:30:44    7.0        42270   Male         Web   
2  2018-01-01  21:55:31   10.0        14563   Male         Web   
3  2018-01-01  13:57:15    9.0        58601   Male         Web   
4  2018-01-01  15:17:41    2.0        48342   Male         Web   

  customer_login_type    product_category                 product  sales  \
0              Member  Auto & Accessories       Car Media Players  140.0   
1              Member  Auto & Accessories  Car Pillow & Neck Rest  231.0   
2              Member  Auto & Accessories            Car Speakers  211.0   
3              Member  Auto & Accessories                    Tyre  250.0   
4              Member  Auto & Accessories                    Tyre  250.0   

   quantity  discount  profit  shipping_cost order_priority payment_method  
0       4.0       0.3    43.2            4.3         Medium       e_wallet  
1       5.0       0.1   139.5           13.9           High    money_order  
2       5.0       0.1   120.5           12.0           High    credit_card  
3       4.0       0.2   150.0           15.0       Critical    credit_card  
4       1.0       0.1   165.0           16.5           High    credit_card

#### Подивись загальну інформацію про датасет. Яка кількість стовпчиків та рядків? Яким є тип даних кожного стовпчика? Чи всі числові стовпчики розпізнаються вірно? Чи є пропущені значення?

In [11]:
sales_data.shape

(51290, 16)

In [9]:
sales_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51290 entries, 0 to 51289
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   order_date           51290 non-null  object 
 1   time                 51290 non-null  object 
 2   aging                51289 non-null  float64
 3   customer_id          51290 non-null  int64  
 4   gender               51290 non-null  object 
 5   device_type          51290 non-null  object 
 6   customer_login_type  51290 non-null  object 
 7   product_category     51290 non-null  object 
 8   product              51290 non-null  object 
 9   sales                51289 non-null  float64
 10  quantity             51288 non-null  float64
 11  discount             51289 non-null  float64
 12  profit               51290 non-null  float64
 13  shipping_cost        51289 non-null  float64
 14  order_priority       51288 non-null  object 
 15  payment_method       51290 non-null 

#### Виведи назви всіх стовпчиків.

In [12]:
sales_data.columns

Index(['order_date', 'time', 'aging', 'customer_id', 'gender', 'device_type',
       'customer_login_type', 'product_category', 'product', 'sales',
       'quantity', 'discount', 'profit', 'shipping_cost', 'order_priority',
       'payment_method'],
      dtype='object')

#### За допомогою методу describe() виведи описову статистику по числових стовпчиках.

In [23]:
sales_data.describe()

aging   customer_id         sales      quantity      discount  \
count  51289.000000  51290.000000  51289.000000  51288.000000  51289.000000   
mean       5.255035  58155.758764    152.340872      2.502983      0.303821   
std        2.959948  26032.215826     66.495419      1.511859      0.131027   
min        1.000000  10000.000000     33.000000      1.000000      0.100000   
25%        3.000000  35831.250000     85.000000      1.000000      0.200000   
50%        5.000000  61018.000000    133.000000      2.000000      0.300000   
75%        8.000000  80736.250000    218.000000      4.000000      0.400000   
max       10.500000  99999.000000    250.000000      5.000000      0.500000   

             profit  shipping_cost  
count  51290.000000   51289.000000  
mean      70.407226       7.041557  
std       48.729488       4.871745  
min        0.500000       0.100000  
25%       24.900000       2.500000  
50%       59.900000       6.000000  
75%      118.400000      11.800000  
max      167.500000      16.800000

#### Висновок

Маємо датасет з 51290 рядків розділених на 16 колонок. 

Серед даних маємо 8 рядків із **NaN**-значенням. Ці рядки можемо знайти за допомогою коду: ```python sales_data[sales_data.isna().any(axis=1)]```. 

Датасет з неправильно визначеними _dtype_ для колонок: order_date, time. Також краще було б віднести ці колонки до [категоріального](https://pandas.pydata.org/docs/user_guide/categorical.html#) типу даних: 
* gender
* device_type
* customer_login_type
* product_category
* order_priority
* payment_method

Для колонок із числовими типами даних аномалій чи дивних значень не виявлено.